src/NetMQ/NetMQPoller.cs

```diff
@@ -243,33 +275,18 @@ public void Remove(ISocketPollable socket)
                 socket.Socket.EventsChanged -= OnSocketEventsChanged;
                 m_sockets.Remove(socket.Socket);
                 m_isPollSetDirty = true;
-            });
+            })
+            .Wait(); 
```

Comment:
Calling `Wait` on tasks can lead to deadlocks in complex code.

I'd suggest making a `RemoveAsync` overload that returns `Task`.

As an aside, it's better to use `GetAwaiter().GetResult()` than `Wait()`, mostly because the latter wraps any thrown exception in a redundant `AggregateException` which can make debugging more difficult. Still, synchronous blocking waits in library code like this should be avoided. There are definitely user scenarios where deadlocks can occur.

Reply:
Lol.

@ first i tried ```await```, but there was an issue using ```await``` in .NET40....  so I changed it to Wait()....  forgot about these methods.

Reply:
The problem likely was that you cannot `await` in a non-`async` method.

Reply:
I think the latest addresses this by adding new `RemoveAsync()` method, and reverting the old version, along with the original issue.
I marked the `Remove` methods as `Obsolete`...